rc/parinfer.kak

```diff
@@ -1,37 +1,77 @@
-declare-option -docstring %{Whether to automatically update the buffer on changes} bool parinfer_enabled yes
+define-command -docstring "parinfer-enable-window [<mode>]: enable Parinfer for current window.
+Modes:
+    -indent  Preserve indentation and fix parentheses (default).
+    -paren   Preserve parentheses and fix indentation.
+    -smart   Try to be smart about what to fix." \
+parinfer-enable-window -params ..1 %{
+    require-module parinfer
+    try %{
+        parinfer -paren
+        # if parinfer -paren fails, we don't set parinfer_enable to true
+        # thus making sure that hooks do not break our code
+        set-option window parinfer_enabled true
+    } catch %{
+        # set up recovery hooks that will re-enable parinfer when parens are balanced
+        hook -group parinfer-try-paren window NormalIdle .* parinfer-try-paren
+        hook -group parinfer-try-paren window InsertIdle .* parinfer-try-paren
+        echo -markup "{Error}parinfer error: %val{error}"
+        echo -debug "parinfer error: %val{error}"
+    }
+    evaluate-commands %sh{
+        mode="${1:-indent}"
+        printf "%s\n" "remove-hooks window parinfer
+                       hook -group parinfer window NormalKey .* %{ parinfer -if-enabled $mode }
```

Comment:
Using `$mode` here leaves things a little funky.... `parinfer_current_mode` no longer allows the user to change the window's mode, and it is set by `parinfer` to indicate the current mode... which means it'll be a lagging indicator, as `parinfer` would have had to run first to be accurate.

I think this should be `%opt{parinfer_current_mode}`, `parinfer` should not set the option, but `parinfer-enable-window` should.

Reply:
Now `parinfer-enable-window` sets `%opt{parinfer_current_mode}`